modules/crud/sql/update.go

```diff
@@ -1,60 +1,226 @@
 package sql
 
 import (
+	"fmt"
 	"context"
 	"strings"
+	"strconv"
 
-	goqu "gopkg.in/doug-martin/goqu.v4"
+	goqu "github.com/doug-martin/goqu/v8"
 
 	_ "github.com/go-sql-driver/mysql"                 // Import for MySQL
 	_ "github.com/lib/pq"                              // Import for postgres
-	_ "gopkg.in/doug-martin/goqu.v4/adapters/postgres" // Adapter for postgres
+	_ "github.com/doug-martin/goqu/v8/dialect/postgres"  // Dialect for postgres
 
 	"github.com/spaceuptech/space-cloud/model"
 	"github.com/spaceuptech/space-cloud/utils"
 )
 
 // Update updates the document(s) which match the condition provided.
 func (s *SQL) Update(ctx context.Context, project, col string, req *model.UpdateRequest) error {
-	sqlString, args, err := s.generateUpdateQuery(ctx, project, col, req)
-	if err != nil {
-		return err
+	return s.update(ctx, project, col, req, s.client)
+}
+
+func (s *SQL) update(ctx context.Context, project, col string, req *model.UpdateRequest, executor interface{}) error {
+	if req == nil {
+		return utils.ErrInvalidParams
 	}
-	return s.doExecContext(ctx, sqlString, args)
+	if req.Update == nil {
+		return utils.ErrInvalidParams
+	}
+	switch req.Operation {
+	case utils.All:
+		for k := range req.Update {
```

Comment:
We shouldn't have a for loop here. `generateUpdateQuery` should internal have this for loop in my opinion.